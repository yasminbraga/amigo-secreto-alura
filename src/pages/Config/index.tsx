import Card from "../../components/Card";
import Footer from "../../components/Footer";
import UserForm from "../../components/UserForm";
import UserList from "../../components/UserList";

const Config: React.FC = () => {
  return (
    <Card title="Vamos comeÃ§ar!">
      <UserForm />
      <UserList />
      <Footer />
    </Card>
  );
};

export default Config;
