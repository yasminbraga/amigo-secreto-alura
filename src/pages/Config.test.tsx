import { render } from "@testing-library/react";
import { RecoilRoot } from "recoil";
import Config from "./Config";

const mockNavigate = jest.fn();

jest.mock("react-router-dom", () => {
  return {
    useNavigate: () => mockNavigate,
  };
});

describe("config page", () => {
  test("must be right rendered", () => {
    const { container } = render(
      <RecoilRoot>
        <Config />
      </RecoilRoot>
    );

    expect(container).toMatchSnapshot();
  });
});
