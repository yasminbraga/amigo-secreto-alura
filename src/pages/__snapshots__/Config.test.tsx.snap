// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config page must be right rendered 1`] = `
<div>
  <div
    class="card"
  >
    <h3
      class="card__title"
    >
      Vamos come√ßar!
    </h3>
    <form
      class="formContainer"
    >
      <input
        class="formContainer__input"
        placeholder="Insira os nomes dos participantes"
        type="text"
        value=""
      />
      <button
        class="formContainer__btn"
        disabled=""
      >
        Adicionar
      </button>
      
    </form>
    <ul
      class="list"
    />
    <footer
      class="footer"
    >
      <button
        class="footer__btn"
        disabled=""
      >
        <img
          alt="play icon"
          src="play.png"
        />
        Iniciar brincadeira!
      </button>
      <img
        alt="sacolas de compras"
        src="sacolas.png"
      />
    </footer>
  </div>
</div>
`;
